app/script/backup/BackupRepository.js

```diff
@@ -132,45 +138,67 @@ z.backup.BackupRepository = class BackupRepository {
   importHistory(archive, initCallback, progressCallback) {
     this.isCanceled = false;
     const files = archive.files;
-    if (!files[BackupRepository.CONFIG.META_FILENAME]) {
+    if (!files[BackupRepository.CONFIG.FILENAME.METADATA]) {
       throw new z.backup.InvalidMetaDataError();
     }
 
-    const verifyMetadataPromise = files[BackupRepository.CONFIG.META_FILENAME]
-      .async('string')
-      .then(JSON.parse)
-      .then(metadata => this.verifyMetadata(metadata));
-
-    const unzipPromise = verifyMetadataPromise.then(() => {
-      return Promise.all(
-        Object.values(archive.files)
-          .filter(zippedFile => zippedFile.name !== BackupRepository.CONFIG.META_FILENAME)
-          .map(zippedFile => zippedFile.async('string').then(value => ({content: value, filename: zippedFile.name})))
-      );
+    return this.verifyMetadata(files)
+      .then(() => this._extractHistoryFiles(files))
+      .then(fileDescriptors => this._importHistoryData(fileDescriptors, initCallback, progressCallback))
+      .catch(error => {
+        this.logger.error(`Failed to import history: ${error.message}`, error);
```

Comment:
We are missing some generic loggers that catch all the errors.
We can easily add them on the `HistoryImportViewModel` and `HistoryExportViewModel` (in the `onError` callback)

Reply:
Will add them there as well.